app/src/visionglass/java/com/igalia/wolvic/PlatformActivity.java

```diff
@@ -126,6 +130,33 @@ protected Intent getStoreIntent() {
     };
     private final Runnable activityResumedRunnable = this::activityResumed;
 
+    // Unfortunately, the usual solutions for keeping the screen active (WakeLock, KEEP_SCREEN_ON)
+    // do not work when the Vision Glass is connected, so we need to simulate a touch periodically.
+    private final Handler mMainHandler = new Handler(Looper.getMainLooper());
+    private final String KEEP_SCREEN_ON_COMMAND = "input touchscreen tap 1 1";
+    private final int KEEP_SCREEN_ON_DELAY = 8_000;
+    private final Runnable keepScreenOnRunnable = new Runnable() {
+        @Override
+        public void run() {
+            if (ProcessLifecycleOwner.get().getLifecycle().getCurrentState().isAtLeast(Lifecycle.State.STARTED)) {
+                try {
+                    Runtime.getRuntime().exec(KEEP_SCREEN_ON_COMMAND);
+                } catch (IOException e) {
+                    Log.e(LOGTAG, "Error while simulating touch", e);
```

Comment:
Nitpicking but I'd use log.i for this, it doesn't seem critical

Reply:
I'll make it a warning and avoid enqueuing more callbacks.